lib/brakeman/checks/check_forgery_setting.rb

```diff
@@ -7,67 +7,69 @@
 class Brakeman::CheckForgerySetting < Brakeman::BaseCheck
   Brakeman::Checks.add self
 
-  @description = "Verifies that protect_from_forgery is enabled in ApplicationController"
+  @description = "Verifies that protect_from_forgery is enabled in direct subclasses of ActionController::Base"
 
   def run_check
-    app_controller = tracker.controllers[:ApplicationController]
-    return unless app_controller and app_controller.ancestor? :"ActionController::Base"
+    tracker.controllers
+        .select { |_, controller| controller.parent == :"ActionController::Base" }
+        .each do |name, controller|
 
-    if tracker.config.allow_forgery_protection?
-      warn :controller => :ApplicationController,
-        :warning_type => "Cross-Site Request Forgery",
-        :warning_code => :csrf_protection_disabled,
-        :message => "Forgery protection is disabled",
-        :confidence => CONFIDENCE[:high],
-        :file => app_controller.file
+      if tracker.config.allow_forgery_protection?
+        warn :controller => name,
```

Comment:
It's warning about a global setting (*that probably no one ever uses*), so it doesn't make sense to warn about it for every controller that inherits from `ActionController::Base`. Actually it doesn't make sense to tie it to a controller at all, but this code was written over five years ago and I didn't know what I was doing.